05_syntax_lab/04.py

```diff
@@ -1,8 +1,10 @@
-"""
-Write a program that reads lines from the user
-until an empty line is inserted.
-After the user typed in an empty line,
-print all previously inserted lines in reverse
-order (from last to first)
-"""
-
+
+line = "dummy"
+arr = []
+index = 0
+while (line != ""):
+    line = raw_input()
+    arr.insert(index, line)
+    index+=1
+for i in range(1,index + 1):
+    print arr[index-i]
```

Comment:
1. Nice! Can you also find a way to solve this without using arrays?

2. While we haven't yet learned arrays, here's still a simpler way to print the reversed list:
```
for item in reversed(arr): print item
```